crates/sncast/src/main.rs

```diff
@@ -670,16 +700,36 @@ async fn run_async_command(cli: Cli, config: CastConfig, ui: &UI) -> Result<()>
         }
 
         Commands::Completions(completions) => {
+            #[derive(Serialize)]
+            struct CompletionsMessage {
+                completions: String,
+            }
+
+            impl Message for CompletionsMessage {
+                fn text(&self) -> String {
+                    self.completions.clone()
+                }
+
+                fn json(&self) -> Value {
+                    json!(self)
+                }
+            }
+
             generate_completions(completions.shell, &mut Cli::command())?;
 
             // TODO(#3560): Remove this warning when the `completion` alias is removed
             if std::env::args().nth(1).as_deref() == Some("completion") {
-                let message = &WarningMessage::new(
+                let message = WarningMessage::new(
                     "Command `sncast completion` is deprecated and will be removed in the future. Please use `sncast completions` instead.",
                 );
 
                 // `#` is required since `sncast completion` generates a script and the output is used directly
-                ui.println(&format!("# {}", message.text()));
+                ui.print_message(
+                    "completion",
+                    CompletionsMessage {
+                        completions: format!("# {}", message.text()),
+                    },
+                );
```

Comment:
It seems a bit artificial to separate a command from its message type. Shouldn't a command always be limited to its message type?

Reply:
Not sure I get it, what change you'd like to have here?

Reply:
I mean that we pass command (in this case `completion`) and the message instance separately, to `print_message`. Why can't the command be placed directly in the message type (as getter or smth)?

Reply:
This doesn't allow for same message to be reused between different commands. Not that we have cases like this now, but it seemed better for me to support these.